packages/transaction-messages/src/__typetests__/instructions-typetests.ts

```diff
@@ -18,6 +21,24 @@ type IInstruction = BaseTransactionMessage['instructions'][number];
         const newMessage = appendTransactionMessageInstruction(null as unknown as IInstruction, message);
         newMessage satisfies BaseTransactionMessage & { some: 1 };
     }
+
+    // It concatenates the instruction types
+    {
+        const message = null as unknown as { instructions: [InstructionA]; version: 0 };
+        const newMessage = appendTransactionMessageInstruction(null as unknown as InstructionB, message);
+        newMessage.instructions satisfies readonly [InstructionA, InstructionB];
+        // @ts-expect-error Wrong order.
+        newMessage.instructions satisfies readonly [InstructionB, InstructionA];
+        // @ts-expect-error Not readonly.
+        newMessage.instructions satisfies [InstructionA, InstructionB];
+    }
+
+    // It adds instruction types to base transaction messages
+    {
+        const message = null as unknown as BaseTransactionMessage;
+        const newMessage = appendTransactionMessageInstruction(null as unknown as InstructionA, message);
+        newMessage.instructions satisfies readonly [...IInstruction[], InstructionA];
```

Comment:
Just wanted to flag this interesting test result. Since `BaseTransactionMessage` already contains a generic instruction array as `IInstruction[]`, we get results like `[...IInstruction[], InstructionA]` when we append. I think this is probably the expected behaviour anyway but there could be a case for having an empty array on the `BaseTransactionMessage` since it starts with no instructions? Not a trivial change though.